lib/Perl/Critic/Policy/Bangs/ProhibitRefProtoOrProto.pm

```diff
@@ -22,14 +22,22 @@ sub violates {
     return if is_subroutine_name($elem);
 
     my $suspectproto = $elem->snext_sibling();
-    if ( $suspectproto && $suspectproto->isa( 'PPI::Token::Symbol' ) ) {
+    if ( $suspectproto && ( $suspectproto->isa( 'PPI::Token::Symbol' ) || $suspectproto->isa( 'PPI::Structure::List' ) ) ) {
         # $suspectproto is the thing I'm calling ref on. Let's see if there's a || after that.
         if ( $suspectproto->snext_sibling()
                 && $suspectproto->snext_sibling->isa( 'PPI::Token::Operator' )
                 && $suspectproto->snext_sibling() eq q{||} ) {
+            my $content;
+            if ( $suspectproto->can( 'schildren' ) ) {
+                my @schildren = $suspectproto->schildren();
+                return unless scalar( @schildren ) == 1 && $schildren[0]->can( 'content' );
+                $content = $schildren[0]->content();
+            } else {
+                $content = $suspectproto->content();
+            }
```

Comment:
I suspect there's a nicer way to do this, but this appears to work at least to some extent.